src/internal/bigbang/bigbang_test.go

```diff
@@ -0,0 +1,296 @@
+			fileName:       "invalid_kind.yaml",
+			expectedOutput: "",
+			expectedErr:    errors.New("values manifests must be a Secret or ConfigMap"),
+		},
+		{
+			name:           "Missing values.yaml",
+			fileName:       "missing_values.yaml",
+			expectedOutput: "",
+			expectedErr:    errors.New("values.yaml key must exist in data"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			filePath := filepath.Join("testdata", "getValuesFromManifest", tt.fileName)
+			output, err := getValuesFromManifest(filePath)
+			if tt.expectedErr != nil {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, tt.expectedOutput, output)
+		})
+	}
+}
+
+func TestAddBigBangManifests(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name          string
+		airgap        bool
+		valuesFiles   []string
+		version       string
+		repo          string
+		expectedFiles []string
+	}{
+		{
+			name:        "Airgap true",
+			airgap:      true,
+			valuesFiles: []string{},
+			version:     "2.35.0",
+			repo:        "https://repo1.dso.mil/big-bang/bigbang",
+			expectedFiles: []string{
+				filepath.Join("testdata", "addBBManifests", "airgap-true", "bb-gitrepository.yaml"),
+				filepath.Join("testdata", "addBBManifests", "airgap-true", "bb-zarf-credentials.yaml"),
+				filepath.Join("testdata", "addBBManifests", "airgap-true", "bb-helmrelease.yaml"),
+			},
+		},
+		{
+			name:   "Airgap false with values files and v2beta1 version",
+			airgap: false,
+			valuesFiles: []string{
+				filepath.Join("testdata", "addBBManifests", "airgap-false", "neuvector.yaml"),
+			},
+			version: "2.0.0",
+			repo:    "https://repo1.dso.mil/big-bang/bigbang",
+			expectedFiles: []string{
+				filepath.Join("testdata", "addBBManifests", "airgap-false", "bb-gitrepository.yaml"),
+				filepath.Join("testdata", "addBBManifests", "airgap-false", "bb-helmrelease.yaml"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tempDir := t.TempDir()
+			var expectedManifests []string
+			for _, f := range tt.expectedFiles {
+				expectedManifests = append(expectedManifests, filepath.Join(tempDir, filepath.Base(f)))
+			}
+			expectedManifests = append(expectedManifests, tt.valuesFiles...)
+			manifest, err := createBBManifests(context.Background(), tt.airgap, tempDir, tt.valuesFiles, tt.version, tt.repo)
+			require.NoError(t, err)
+			require.ElementsMatch(t, expectedManifests, manifest.Files)
+
+			for _, expectedFile := range tt.expectedFiles {
+				_, filename := filepath.Split(expectedFile)
+				generatedFile := filepath.Join(tempDir, filename)
+				expectedContent, err := os.ReadFile(expectedFile)
+				require.NoError(t, err)
+				generatedContent, err := os.ReadFile(generatedFile)
+				require.NoError(t, err)
+				require.Equal(t, string(expectedContent), string(generatedContent))
+			}
+		})
+	}
+}
+
+func TestCreate(t *testing.T) {
```

Comment:
This is more similar to an e2e test. It calls out to the internet many times to download several different repos and files. It would be hard and time consuming to mock all of the internet calls. 

 I can move this test to e2e tests, not sure how we feel about it.